components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/RequestObjectValidatorImpl.java

```diff
@@ -74,14 +76,101 @@ public boolean isSigned(RequestObject requestObject) {
     public boolean validateSignature(RequestObject requestObject, OAuth2Parameters oAuth2Parameters) throws
             RequestObjectException {
 
+        boolean isVerified;
+        Certificate certificate = null;
         SignedJWT jwt = requestObject.getSignedJWT();
-        Certificate certificate =
-                getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
-        boolean isVerified = isSignatureVerified(jwt, certificate);
+        try {
+             certificate =
+                    getCertificateForAlias(oAuth2Parameters.getTenantDomain(), oAuth2Parameters.getClientId());
+        } catch (RequestObjectException e) {
+            log.warn("Error retrieving public certificate for service provider checking whether a jwks endpoint " +
+                    "is configured for the service provider ");
+            if (log.isDebugEnabled()) {
+
+                log.debug("Error retrieving public certificate for service provider checking whether a jwks endpoint " +
+                        "is configured for the service provider: " + e);
```

Comment:
Include who the service provider is. As now you have used the same string to log the warn and the debug log better if you can initialize that string message and use it in both

Reply:
done